packages/core/src/Credentials.ts

```diff
@@ -23,6 +23,11 @@ export class Credentials extends ICredentials {
 	 * Sets new credential object
 	 */
 	setData(data: ICredentialDataDecryptedObject): void {
+		// when this function is being called via the object prototype, cipher is undefined
+		if (!this.cipher) {
```

Comment:
can we update the codebase to not use `prototype.setData.call` instead ?

Reply:
I assumed you made that change for a reason?

Reply:
[looks like it was added forever ago](https://github.com/n8n-io/n8n/commit/09d9e12db9534f272c8bbd9ed0f2a374e0cab180#diff-f59acdc6789ce3cf6619c3132d5e39545f06b3de375352c37380a72fb792b322R72), and we just hadn't noticed it so far.